service/history/execution/mutable_state_decision_task_manager.go

```diff
@@ -254,7 +253,8 @@ func (m *mutableStateDecisionTaskManagerImpl) ReplicateDecisionTaskCompletedEven
 	event *types.HistoryEvent,
 ) error {
 	m.beforeAddDecisionTaskCompletedEvent()
-	return m.afterAddDecisionTaskCompletedEvent(event, math.MaxInt32)
+	domainName := m.msb.GetDomainEntry().GetInfo().Name
+	return m.afterAddDecisionTaskCompletedEvent(event, m.msb.config.MaxAutoResetPoints(domainName))
```

Comment:
Does this need a nil check? I'm a bit worried that this config value isn't set in some circumstances?

Reply:
It is used in multiple places, but yeah, seems we don't have nil check in these funcs. I can try writing unit test here, this file is not covered at all :(

Reply:
Added  nil check here. Seems it is also safe without check, we build the domainEntry from the very base level